views/js/pciCreator/ims/mathEntryInteraction/creator/widget/states/Map.js

```diff
@@ -73,48 +78,136 @@ define([
-                    }
+    MathEntryInteractionStateResponse.prototype.initResponseForm = function initResponseForm() {
+        if (this.correctResponses.length > 0) {
+            return false;
+        }
+        var interaction = this.widget.element;
 
-                    newCorrectAnswer = newCorrectAnswer.join(',');
-                } else {
-                    newCorrectAnswer = '';
+        var newCorrectAnswer;
+
+        if (this.inGapMode() === true) {
+            this.emptyGapFields();
+            var gapExpression = interaction.prop('gapExpression');
+            var gapCount = (gapExpression.match(/\\taoGap/g) || []).length;
+            if (gapCount > 0) {
+                newCorrectAnswer = [];
+                for (var i = 0; i < gapCount; i++) {
+                    newCorrectAnswer.push(' ');
                 }
+
+                newCorrectAnswer = newCorrectAnswer.join(',');
             } else {
                 newCorrectAnswer = '';
-                self.toggleResponseMode(false);
             }
+        } else {
+            newCorrectAnswer = '';
+            this.toggleResponseMode(false);
+        }
+
+        this.correctResponses.push(newCorrectAnswer);
+    }
 
-            self.correctResponses.push(newCorrectAnswer);
-            self.renderForm(self.correctResponses);
+    MathEntryInteractionStateResponse.prototype.createScoreResponse = function createScoreResponse() {
+        var $container = this.widget.$container;
+        var $input = $container.find('.math-entry-input');
+
+        if ($container.find('.math-entry-response-wrap').length > 0) {
+            return false
+        }
+
+        var $interaction = this.widget.element;
+        var response = $interaction.getResponseDeclaration();
+        this.$responseCorrectTitle = $('<span>', {
+            'class': 'math-entry-response-title math-entry-response-correct'
+        }).html('Correct');
+        $input[0].parentNode.prepend(this.$responseCorrectTitle[0]);
+        $input[0].parentNode.className = 'math-entry-correct-wrap';
+        $input[0].parentNode.dataset.index = 0;
+        this.$scoreDiv = $('<div>', {
+            'class': 'math-entry-score-wrap math-entry-response-correct'
         });
+        $input[0].parentNode.parentNode.insertBefore(this.$scoreDiv[0], $input.nextSibling);
+        this.$scoreDiv.after($addAlternativeBtn);
+        $('.math-entry-correct-wrap, .math-entry-score-wrap').wrapAll('<div class="math-entry-response-wrap"></div>');
+        var pairId = $interaction.attr('responseIdentifier');
+    
+        this.$responseCorrectScore = $('<span>', {
+            'class': 'math-entry-score-title math-entry-response-correct'
+        }).html('Score');
+        this.$scoreDiv.append(scoreTpl({
+            serial: response.serial,
+            mathIdentifier: pairId,
+            placeholder: response.getMappingAttribute('defaultValue')
+        }));
+        this.$scoreDiv.prepend(this.$responseCorrectScore);
```

Comment:
**issue:** We must not have inline HTML in the code. It should go to a template.
Moreover, this code looks verbose and error-prone. It should be simplified.